Userland/Libraries/LibJS/Tests/time-from-year.js

```diff
@@ -0,0 +1,11 @@
+test("YearFromTime(TimeFromYear(t)) <= t", () => {
```

Comment:
Instead of adding this as a random test file in the root LibJS/Tests directory, let's add it to LibJS/Tests/builtins/Date/Date.UTC.js (with a plain-english test name too please - if I came across this test a year from now, I'd have no idea what's special about the date being tested here).

Reply:
I settled for `month from negative time` as the test title. It is a bit difficult to choose it, since what that input uncovered is precisely that the condition from `YearFromTime(TimeFromYear(t)) <= t` wasn't being held in the case of negative time.

Since that condition didn't hold, the interpreter crashed before it could get out of the `Date` constructor.

Reply:
I settled for `month from negative time` as the test title. It is a bit difficult to choose it, since what that input uncovered is precisely that the condition from `YearFromTime(TimeFromYear(t)) <= t` wasn't being held in the case of negative time.

Since that condition didn't hold, the interpreter crashed before it could get out of the `Date` constructor.